api/agent/agent.go

```diff
@@ -880,9 +880,9 @@ func (a *agent) runHot(ctx context.Context, caller slotCaller, call *call, tok R
 	if tryQueueErr(err, errQueue) != nil {
 		return
 	}
-
 	needsPull, err := cookie.ValidateImage(ctx)
 	if needsPull {
+		call.PullStartedAt = common.DateTime(time.Now())
```

Comment:
This is tricky here. If multiple pulls are being executed for the same image, then this metric here will not yield the true docker pull duration, but it will simply show **the amount of wait time for this consumer code.**   Basically, we have 2 metrics at work here, 1) true docker-pull time 2) wait time experienced here in this code-block.

For example, another client might have already triggered a docker-pull a few seconds earlier. If this code blocks on that as well, then the wait time here would show less time than the actual docker-pull duration, since earlier client started the download earlier.

If we are trying to account for how much time this client code have waited, then the metric here as is is OK and good. But in that case, maybe we should rename this metric as 'docker-pull-wait'. I think this is what we should do... I suggest a small rename to clarify that this metric here may not be the true download duration.

Reply:
The true pull duration is what we need, not the pull-wait as you suggested, let me revise this. I'm thinking maybe the method used to add userexecution time, by incrementing using time.Duration would be better

Reply:
https://github.com/fnproject/fn/blob/8cef854cccf96da4d78cf1aa6d0dd8b478a8fe8b/api/agent/pure_runner.go#L239

https://github.com/fnproject/fn/blob/8cef854cccf96da4d78cf1aa6d0dd8b478a8fe8b/api/agent/agent.go#L899

Here you go, docker pull wait now